lib/server.js

```diff
@@ -816,7 +816,7 @@ Server.prototype._onRequest = function _onRequest(req, res) {
         handlerDomain.add(req);
         handlerDomain.add(res);
         handlerDomain.on('error', function onError(err) {
-            self._onHandlerError(err, req, res);
+            self._onHandlerError(err, req, res, true);
```

Comment:
I think this would still make errors _emitted_ on objects bound to the domain (in this case `req` and `res`) trigger an `uncaughtException` event. It seems that would also be a problem?

If we consider that is indeed not what the behavior of `handleUncaughtExceptions` should be, we should set the fourth parameter of `_onHandlerError` to `true` only if `err.domainThrown === true`.

@mridgway @DonutEspresso Thoughts?

Reply:
I believe that behavior is consistent with Restify versions <7. I'm not sure what the original intention was though.

I agree that it would make more sense to filter to only uncaught exceptions.

Reply:
> I believe that behavior is consistent with Restify versions <7.

I just verified that looking at the 6.x branch of this repository and this is correct. Thus I'm fine with keeping that behavior, at least for now.